### PR TITLE
man: say more explicitly what to do in the SD_JOURNAL_INVALIDATE event

### PR DESCRIPTION
**Pull request by poettering**
This rewords the section, explicitly distuingishing the cases of clients
that only want a continious log stream (which can simply treat
SD_JOURNAL_INVALIDATE the same way as SD_JOURNAL_APPEND) and those which
want to represent on screen the full state of the log data on disk.

This is an alternative to a part of PR #9060, but keeps an explanation
of the destinction of handling depending on the type of client.

Fixes: #8963

### CODE REVIEW

File: man/sd_journal_get_fd.xml [hunk 1]
```diff
@@ -199,22 +199,21 @@ else {
...
-    sufficient to simply continue reading at the previous end of the
-    journal.</para>
+    <para><function>sd_journal_process()</function> and <function>sd_journal_wait()</function> return one of
+    <constant>SD_JOURNAL_NOP</constant>, <constant>SD_JOURNAL_APPEND</constant> or
+    <constant>SD_JOURNAL_INVALIDATE</constant> on success or a negative errno-style error code. If
+    <constant>SD_JOURNAL_NOP</constant> is returned, the journal did not change since the last invocation. If
+    <constant>SD_JOURNAL_APPEND</constant> is returned, new entries have been appended to the end of the journal.  In
+    the case of <constant>SD_JOURNAL_APPEND</constant>, it is sufficient to simply continue reading at the previous end
+    location of the journal, to read the newly added entries. If <constant>SD_JOURNAL_INVALIDATE</constant>, journal
+    files were added to or removed from the current view on the journal stream (possibly due to rotation or vacuuming),
```
Comment (keszybz) [inline man/sd_journal_get_fd.xml]: what is "current view"? I thought INVALIDATE is returned if any journal files in the relevant directories are added or removed, even if they wouldn't be shown because of installed matches.

File: man/sd_journal_get_fd.xml [hunk 2]
```diff
@@ -199,22 +199,21 @@ else {
...
+    and thus entries might have appeared or disappeared at arbitrary places in the log stream, possibly before or after
+    the previous end of the log stream. If <constant>SD_JOURNAL_INVALIDATE</constant> is returned live-view UIs that
```
Comment (keszybz) [inline man/sd_journal_get_fd.xml]: returned, comma, live-view UIs.

File: man/sd_journal_get_fd.xml [hunk 3]
```diff
@@ -199,22 +199,21 @@ else {
...
+    want to reflect on screen the precise state of the log data on disk should probably refresh their entire display
+    (relative to the cursor of the log entry on the top of the screen). Programs only interested in a continous stream
+    of log data as it happens may treat <constant>SD_JOURNAL_INVALIDATE</constant> the same way as
```
Comment (keszybz) [inline man/sd_journal_get_fd.xml]: So "continuous" and "as is happens" are wrong here. If I am browsing logs from last week like with `journalctl -b-2`, the appropriate response to INVALIDATE is what journalctl does, i.e. just continue reading, exactly the same as after APPEND.

File: man/sd_journal_get_fd.xml (hunk 1 above)
Comment (poettering) [reply man/sd_journal_get_fd.xml]: hmm, ok, by "view" i meant SD_JOURNAL_SYSTEM vs. SD_JOURNAL_CURRENT_USER vs. SD_JOURNAL_RUNTIME_ONLY and so on. lemme reword this

File: man/sd_journal_get_fd.xml (hunk 3 above)
Comment (poettering) [reply man/sd_journal_get_fd.xml]: ok, i'll reword this to "strictly sequencial", which i think describes this better?

**Comment by poettering** [comment]
Force puhed a new version. I addressed the points raised. I also converted this into an itemized list, to give this a more clear structure. PTAL!
